apps/generator/lib/conditionalGeneration.js

```diff
@@ -0,0 +1,269 @@
+const fs = require('fs');
+const path = require('path');
+const Ajv = require('ajv');
+const log = require('loglevel');
+const logMessage = require('./logMessages');
+const jmespath = require('jmespath');
+/**
+ * Determines if file generation should be skipped based on templateConfig conditions.
+ *
+ * @param  {AsyncAPIDocument} asyncapiDocument AsyncAPI document to use for condition evaluation.
+ * @param  {String} relativeSourceFile Relative path of the source file.
+ * @param  {String} relativeSourceDirectory Relative path of the source directory.
+ * @param  {String} relativeTargetFile Relative path of the target file to be generated.
+ * @param  {Object} templateConfig Template configuration.
+ * @param  {String} targetDir Directory where the generated files are written.
```

Comment:
there are more arguments in function that you have here in jsdock

Reply:
Resolved thanks